tests/scripts/generate_test_keys.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+"""Module generating EC and RSA keys to be used in test_suite_pk instead of
+generating the required key at run time. This helps speeding up testing."""
+
+import os
+import sys
+import subprocess
+
+KEY_GEN = "./programs/pkey/gen_key"
```

Comment:
Using `gen_key` is problematic for several reasons:

* You need to run an untested build of Mbed TLS before you can generate the test data.
* Since the data comes from Mbed TLS, we are only validating that Mbed TLS is compatible with itself, not that it behaves correctly.
* This is randomized, so the results are not reproducible.

I'm hesitating between requesting follow-ups to resolve these downsides, and rejecting this approach outright.

Reply:
> I'm hesitating between requesting follow-ups to resolve these downsides, and rejecting this approach outright.

This script disappears in #9010 so it won't stay around for too long. In the worst case I can kill this PR and continue with only #9010. As you prefer.